src/agni/server/manager/UserManager.java

```diff
@@ -16,8 +16,69 @@ public UserManager(InfoSender infoSender,
     }
 
     @Override
-    public void userRequest(String ip, byte type) {
+    public void userRequest(String ip, byte type, String message) {
         // TODO Auto-generated method stub
+    	String username = userDataGuard.getUsername(ip); 
+        if(type == 0x00) { 			// join chat; message = group to join
```

Comment:
Instead of keeping these in bytes, we should store the bytes as enums so it's easier to debug later. Here's code for the enum.

```java
    public enum UserRequestType {
        JOIN_CHAT((byte)0x00),
        LEAVE_CHAT((byte)0x01),
        FRIENDS_LIST((byte)0x02);
        FRIEND_STATUS((byte)0x03);
        ADD_FRIEND((byte)0x04);
        LOGOUT((byte)0x05);

        private final byte bytes;
        private Status(byte bytes) {
            this.bytes = bytes;
        }

        public final byte bytes() {
            return bytes;
        }
    }
```